NeoMathEngine/src/GPU/CUDA/Kernels/CudaVectorMathKernels.h

```diff
@@ -167,6 +192,61 @@ __global__ void VectorSumAlongDimensionKernel( const float* __restrict__ input,
 	}
 }
 
+__global__ void VectorCumSumAlongDimensionKernel( const float* __restrict__ input, int precedingDims, int dims,
+	int followingDims, float* result )
+{
+	int x;
+	int y;
+	if( GetCudaTaskIndex2D( precedingDims, followingDims, x, y ) ) {
+		const int offset = y * dims * precedingDims + x;
+		input += offset;
+		result += offset;
+		*result = *input;
+		for( int i = 1; i < dims; i++ ) {
+			input += precedingDims;
+			*(result + precedingDims) = *result + *input;
```

Comment:
(minor optimization):
```
float curSum = *input;
*result = currSum;
for( int i = ... ) {
    input += precedingDims;
    result += precidingDims;
    currSum += *input;
    *result = currSum;
}
```
Reason: each call to `*result` or `*input` is a potential access to the global memory, which is slow.